sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/ActionDescriptorFactory.scala

```diff
@@ -48,11 +52,35 @@ private[impl] object ActionDescriptorFactory extends ComponentDescriptorFactory
           .withKalixOptions(kalixOptions)
       }
 
+    val subscriptionTopicMethods = component.getMethods
+      .filter(hasTopicSubscription)
+      .sorted // make sure we get the methods in deterministic order
+      .map { method =>
+        val subscriptionOptions = eventingInForTopic(method)
+        val kalixOptions =
+          kalix.MethodOptions.newBuilder().setEventing(subscriptionOptions).build()
+
+        KalixMethod(RestServiceMethod(method))
+          .withKalixOptions(kalixOptions)
+      }
```

Comment:
Should probably validate that no two method accept the same event as parameter and subscribe to the same topic (since routing is done based on event type)

Reply:
Good idea. 

We have another related issue that came up in our discussion  today. See https://github.com/lightbend/kalix/issues/7407

Reply:
Shall we take into account that the user may want to do two different things in two different methods both pointing to the same topic and consuming the same type of object?